### PR TITLE
#7: Proof of concept for the test suite

### PR DESCRIPTION
**Pull request by jpoly1219**
Created a simple test suite for the `erase_exp` function. More tests need to be added, but this PR aims to be a proof of concept. Please let me know what your thoughts are and what you'd like to see me fix. Thank you!

The imports will likely change after I wrap hazelnut types in modules.

### CODE REVIEW

File: test/Test_erase_exp.re [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+};
+
+let test_eenehole = () => {
+  let ze: zexp = NEHole(Cursor(Lam("f", Lit(1))));
+  let given: hexp = erase_exp(ze);
+  let expected: hexp = NEHole(Lam("f", Lit(1)));
+  check(hexp_typ, "same hexp", given, expected);
+};
+
+let erase_exp_tests = [
```
Comment (cyrus-) [inline test/Test_erase_exp.re:144]: maybe add a few tests where the cursor is more than one level deep to make sure the student isn't pattern matching just one or two levels down and is recursing correctly.

**Review by cyrus- — CHANGES_REQUESTED** [review]
in addition to the small comment, you should add appropriate instructions to the README for running the tests

File: test/Test_erase_exp.re (hunk 1 above)
Comment (jpoly1219) [reply test/Test_erase_exp.re:144]: Sure thing, I'll fix it 👍

**Comment by jpoly1219** [comment]
This review request also handles #3 as well.

File: README.md [hunk 2]
```diff
@@ -71,6 +71,11 @@ To erase the build:
 make clean
 ```
 
+To test your implementation:
+```sh
+make test_impl
```
Comment (cyrus-) [inline README.md]: just call it `make test`

File: README.md [hunk 3]
```diff
@@ -112,6 +117,10 @@ If anything unexpected happens, a warning will appear at the bottom. Here's what
 | Unimplemented     | You called upon an operation that you haven't implemented yet.                                                                                         |
 | Theorem violation | Your implementation has a bug that caused it to violate a Hazelnut metatheorem. Note: This won't catch every metatheorem violation, only some of them. |
 
+## Testing your implementation
+
+We include a test suite to test your implementation. Run `make test_impl` at the root of the project directory to run the test suite.
```
Comment (cyrus-) [inline README.md]: Can remove above mention of `make test` since its mentioned here.

**Review by cyrus- — CHANGES_REQUESTED** [review]
Looking good! In addition to the small README tweak, let's go ahead and write some more tests. Maybe you can recruit other folks who have done diy-hazelnut recently to help write tests (feel free to post in #learning or I'm happy to do that if you prefer.)

**Comment by jpoly1219** [comment]
I would honestly prefer that you write the tests because I think it would be a lot more robust :) And I would also like to see how a more experienced OCaml/Reason dev would write code. I will write more tests and define more interface functions, but progress will be slow until after the exams this week (which happens to go on until Friday).

**Comment by cyrus-** [comment]
@jpoly1219 I don't think I'll have time to write a bunch of tests soon, so let's recruit folks on Slack to help. I'll make a post.

**Review by cyrus- — APPROVED** [review]
Approving this to get the testing infrastructure merged, but we should write a more extensive test suite in a subsequent PR. Thanks @jpoly1219!

**Pull request merged**